OpenTrons/opentrons-api
# feat(components, protocol-designer): wire up liquid toolbox

# Overview

Create expandable liquid cards with enumerated full wells and volumes for each well.

Closes AUTH-754

## Test Plan and Hands on Testing

- Create or upload a protocol and add labware to the deck
- Select labware > 'Add liquid'
- Ensure you've created at several liquids to test with
- Select wells of labware and add different liquids in various volumes to different wells
- Verify that 'Add liquid' input only shows if wells are selected
- Verify that expandable liquid cards show, and when expanded, enumerated wells are in order and show correct volumes

TODO:
- Replace app `Divider` component with new components library version (?)

## Changelog

- wire up liquid cards in `LiquidToolbox`
- create new instance of `Divider` component for use across apps
- update color of `ListItem`
- create liquid card component
- add translations
- fix tests

## Review requests

see test plan and check liquid hashing logic

## Risk assessment

Low
